UDT.Business/Implementation/SubjectService.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using UDT.Business.Interfaces;
+using UDT.Model.Entities;
+using UDT.Repository;
+
+namespace UDT.Business.Implementation
+{
+    public class SubjectService : ISubjectService
+    {
+        private readonly DataContext _dataContext;
+
+        public SubjectService(DataContext dataContext)
+        {
+            _dataContext = dataContext;
+        }
+
+        public IAsyncEnumerable<Subject> GetAllAsync()
+        {
+            return _dataContext.Subjects
+                .AsAsyncEnumerable();
+        }
+
+        public async Task<Subject> GetByIdAsync(int id)
+        {
+            return await _dataContext.Subjects
+                .Include(subject => subject.Users)
+                .FirstOrDefaultAsync(subject => subject.Id == id);
+        }
+
+        public async Task<Subject> AddAsync(Subject subject)
+        {
+            for (var i = 0; i < subject.Users.Count; i++)
```

Comment:
nu sunt sigur daca am inteles exact ce e cu for-ul asta, dar eu cred ca mai bine adaugi obiectul direct in db decat sa faci apel la db pt fiecare user, iar daca nu exista userul respectiv o sa arunce eroare; trebe doar sa vezi ce arunca si sa o prinzi, si atunci faci direct add la subject